vendors/nuvoton/sdk/middleware/wifi_esp8266/esp8266_wifi.c

```diff
@@ -17,8 +17,19 @@
 /* Wi-Fi driver includes. */
 #include "esp8266_wifi.h"
 
+#define ESP8266_ALL_SOCKET_IDS      0xFF
```

Comment:
Hi @cyliangtw,

I noticed some build warnings from this file. Are these warnings present for you as well?

```
[46/228] Building C object CMakeFiles\afr_wifi.dir\vendors\nuvoton\sdk\middleware\wifi_esp8266\esp8266_wifi.o
"..\..\..\vendors\nuvoton\sdk\middleware\wifi_esp8266\esp8266_wifi.c", line 364: Warning:  #175-D: subscript out of range
                  xWifiIpd[2].IsOpen = pdFALSE;
                  ^
"..\..\..\vendors\nuvoton\sdk\middleware\wifi_esp8266\esp8266_wifi.c", line 368: Warning:  #175-D: subscript out of range
                  xWifiIpd[3].IsOpen = pdFALSE;
                          ^
"..\..\..\vendors\nuvoton\sdk\middleware\wifi_esp8266\esp8266_wifi.c", line 372: Warning:  #175-D: subscript out of range
                  xWifiIpd[4].IsOpen = pdFALSE;
                          ^
"..\..\..\vendors\nuvoton\sdk\middleware\wifi_esp8266\esp8266_wifi.c", line 404: Warning:  #1293-D: assignment in condition
              if (pcRecv = strstr(pcPtr, AT_PASSIVE_STRING)) {
                  ^
"..\..\..\vendors\nuvoton\sdk\middleware\wifi_esp8266\esp8266_wifi.c", line 441: Warning:  #1293-D: assignment in condition
              } else if (pcRecv = strstr(pcPtr, AT_RECV_STRING)) {
                         ^
"..\..\..\vendors\nuvoton\sdk\middleware\wifi_esp8266\esp8266_wifi.c", line 548: Warning:  #177-D: variable "usCount"  was declared but never referenced
      uint16_t usCount;
               ^
"..\..\..\vendors\nuvoton\sdk\middleware\wifi_esp8266\esp8266_wifi.c", line 1069: Warning:  #177-D: variable "ucExit"  was declared but never referenced
      uint8_t ucExit = 0;
              ^
"..\..\..\vendors\nuvoton\sdk\middleware\wifi_esp8266\esp8266_wifi.c", line 1152: Warning:  #177-D: variable "xTickCurrent"  was declared but never referenced
      TickType_t xTickCurrent, xTickTimeout;
                 ^
"..\..\..\vendors\nuvoton\sdk\middleware\wifi_esp8266\esp8266_wifi.c", line 1152: Warning:  #177-D: variable "xTickTimeout"  was declared but never referenced
      TickType_t xTickCurrent, xTickTimeout;
                               ^
"..\..\..\vendors\nuvoton\sdk\middleware\wifi_esp8266\esp8266_wifi.c", line 1153: Warning:  #177-D: variable "ucDoFlag"  was declared but never referenced
      uint8_t ucDoFlag = 1;
              ^
..\..\..\vendors\nuvoton\sdk\middleware\wifi_esp8266\esp8266_wifi.c: 10 warnings, 0 errors
```

Reply:
Hi @muneebahmed10 ,
I built by Keil IDE, there are only 2 warnings:
```
compiling aws_dev_mode_key_provisioning.c...
compiling esp8266_wifi.c...
../../../../../vendors/nuvoton/sdk/middleware/wifi_esp8266/esp8266_wifi.c(404): warning:  #1293-D: assignment in condition
              if (pcRecv = strstr(pcPtr, 
T_PASSIVE_STRING)) {
../../../../../vendors/nuvoton/sdk/middleware/wifi_esp8266/esp8266_wifi.c(441): warning:  #1293-D: assignment in condition
              } else if (pcRecv = strstr(pcPtr, 
T_RECV_STRING)) {
../../../../../vendors/nuvoton/sdk/middleware/wifi_esp8266/esp8266_wifi.c: 2 warnings, 0 errors
```

Reply:
Hi @muneebahmed10,

I adjust the default max sockets as 5 to fulfill current wifi module and avoid the warnings #175-D in commit #175-D.

Reply:
Hi @cyliangtw,

Can you take a look at the lines for the unused variable warnings and fix those those as well? Most of them do seem to be unused.

Reply:
Hi @muneebahmed10,
I removed unused variable warnings #177-D in commit 208c3b4 & c3912ef.
Other #177-D are in 3rd-party lib or free-rtos kernel, it could be suppressed by "--diag_suppress=" in Keil IDE.